crates/cli/commands/src/stage/drop.rs

```diff
@@ -78,6 +78,11 @@ impl<C: ChainSpecParser<ChainSpec: EthChainSpec + EthereumHardforks>> Command<C>
             StageEnum::Bodies => {
                 tx.clear::<tables::BlockBodyIndices>()?;
                 tx.clear::<tables::Transactions>()?;
+                if let Some(mut prune_checkpoint) =
+                    tx.get::<tables::PruneCheckpoints>(PruneSegment::Transactions)?
```

Comment:
this part can go inside the new function body too, no? changing the function name to `update_prune_state` would be appropriate for the case where the prune checkpoint is some and is set, as well as the case when the prune checkpoint is none